packages/suite/src/actions/suite/notificationActions.ts

```diff
@@ -1,41 +1,59 @@
 import * as React from 'react';
 import { FormattedMessage } from 'react-intl';
-import { CallbackAction } from '@wallet-reducers/notificationReducer';
 import { NotificationProps } from '@trezor/components';
+import { CallbackAction } from '@suite/reducers/suite/notificationReducer';
 import { NOTIFICATION } from './constants';
 import { GetState, Dispatch, TrezorDevice } from '@suite-types';
 
-interface NotificationPayload {
-    variant: NotificationProps['variant'];
+export interface NotificationAddPayload {
+    variant: NonNullable<NotificationProps['variant']>;
+    id?: string;
+    devicePath?: string;
     title: React.ReactNode | FormattedMessage.MessageDescriptor;
     message?: React.ReactNode;
     cancelable: boolean;
     actions?: CallbackAction[];
 }
 
+interface CloseByKey {
+    key: string;
+    id?: string;
+    devicePath?: string;
+}
+interface CloseById {
+    key?: string;
+    id: string;
+    devicePath?: string;
+}
+interface CloseByDevicePath {
+    key?: string;
+    id?: string;
+    devicePath: string;
+}
+
+export type NotificationClosePayload = CloseByDevicePath | CloseById | CloseByKey;
```

Comment:
I had a little bit of fun with TS and tried to rewrite it without the need for 3 almost-the-same interfaces
```
interface CloseProps {
    key?: string;
    id?: string;
    devicePath?: string;
}

type RequiredKey<M, K extends keyof M> = Omit<M, K> & Required<Pick<M, K>>;
export type NotificationClosePayload =
    | RequiredKey<CloseProps, 'devicePath'>
    | RequiredKey<CloseProps, 'id'>
    | RequiredKey<CloseProps, 'key'>;
```

Reply:
Nice, looks good.